packer/virtualbox.pkr.hcl

```diff
@@ -0,0 +1,336 @@
+packer {
+  required_version = ">= 1.7.0, < 2.0.0"
+
+  required_plugins {
+    virtualbox = {
+      source  = "github.com/hashicorp/virtualbox"
+      version = ">= 1.0.0, < 2.0.0"
+    }
+  }
+}
+
+variable "boot_wait" {
+  type    = string
+  default = "1s"
+}
+
+variable "bundle_iso" {
+  type    = string
+  default = "false"
+}
+
+variable "cpus" {
+  type    = string
+  default = "1"
+}
+
+variable "disk_size" {
+  type    = string
+  default = "50000"
+}
+
+variable "guest_os_type" {
+  type    = string
+  default = "Ubuntu_64"
+}
+
+variable "headless" {
+  type    = string
+  default = "false"
+}
+
+variable "host_port_max" {
+  type    = string
+  default = "4444"
+}
+
+variable "host_port_min" {
+  type    = string
+  default = "2222"
+}
+
+variable "http_port_max" {
+  type    = string
+  default = "9000"
+}
+
+variable "http_port_min" {
+  type    = string
+  default = "8000"
+}
+
+variable "iso_checksum" {
+  type    = string
+  default = "sha256:5e38b55d57d94ff029719342357325ed3bda38fa80054f9330dc789cd2d43931"
+}
+
+variable "iso_file" {
+  type    = string
+  default = "ubuntu-22.04.2-live-server-amd64.iso"
+}
+
+variable "iso_path_external" {
+  type    = string
+  default = "https://releases.ubuntu.com/releases/jammy"
+}
+
+variable "keep_registered" {
+  type    = string
+  default = "false"
+}
+
+variable "memory" {
+  type    = string
+  default = "2048"
+}
+
+variable "packer_cache_dir" {
+  type    = string
+  default = "${env("PACKER_CACHE_DIR")}"
+}
+
+variable "shutdown_timeout" {
+  type    = string
+  default = "30m"
+}
+
+variable "skip_export" {
+  type    = string
+  default = "false"
+}
+
+variable "ssh_agent_auth" {
+  type    = string
+  default = "false"
+}
+
+variable "ssh_clear_authorized_keys" {
+  type    = string
+  default = "true"
+}
+
+variable "ssh_disable_agent_forwarding" {
+  type    = string
+  default = "false"
+}
+
+variable "ssh_file_transfer_method" {
+  type    = string
+  default = "scp"
+}
+
+variable "ssh_handshake_attempts" {
+  type    = string
+  default = "100"
+}
+
+variable "ssh_keep_alive_interval" {
+  type    = string
+  default = "5s"
+}
+
+variable "ssh_username" {
+  type    = string
+  default = "ubuntu"
+}
+
+variable "ssh_password" {
+  type    = string
+  default = "ubuntu"
+}
+
+variable "ssh_port" {
+  type    = string
+  default = "22"
+}
+
+variable "ssh_pty" {
+  type    = string
+  default = "false"
+}
+
+variable "ssh_timeout" {
+  type    = string
+  default = "60m"
+}
+
+variable "start_retry_timeout" {
+  type    = string
+  default = "5m"
+}
+
+variable "version" {
+  type    = string
+  default = "0.0.0"
+}
+
+variable "vm_name" {
+  type    = string
+  default = "studio-selfhosted"
+}
+
+variable "vnc_vrdp_bind_address" {
+  type    = string
+  default = "127.0.0.1"
+}
+
+variable "vnc_vrdp_port_max" {
+  type    = string
+  default = "6000"
+}
+
+variable "vnc_vrdp_port_min" {
+  type    = string
+  default = "5900"
+}
```

Comment:
Other than variables such as `iso_file`, we're not likely to change any these in the future, right?

If so, let's just pass the strings directly in the
`source "virtualbox-iso" "vbox" {`
block

This will make the file much easier to read

Reply:
Got it. 
But let's do a follow-up PR where we hardcode variables like `ssh_handshake_attempts`. There's no reason to have these as variables. We can keep the ones for the Ubuntu version, shasum, and so on.